lib/TransportRPCClient.js

```diff
@@ -5,9 +5,11 @@ const uuid = require('uuid')
 const TransportRPC = require('./TransportRPC')
 
 class TransportRPCClient extends TransportRPC {
-  constructor (client, conf) {
+  constructor (client, conf, Socket) {
```

Comment:
@robertkowalski may I ask you why you pass the class here and not directly the instance?
Having it as a parameter (uppercase) in the constructor is not striking me like smooth.